Lib/sysconfig.py

```diff
@@ -545,15 +545,32 @@ def _init_non_posix(vars):
     """Initialize the module as appropriate for NT"""
     # set basic install directories
     import _imp
+    import re
     vars['LIBDEST'] = get_path('stdlib')
     vars['BINLIBDEST'] = get_path('platstdlib')
     vars['INCLUDEPY'] = get_path('include')
-    try:
+
+    extension_suffixes = _imp.extension_suffixes()
+    if len(extension_suffixes) >= 1:
         # GH-99201: _imp.extension_suffixes may be empty when
         # HAVE_DYNAMIC_LOADING is not set. In this case, don't set EXT_SUFFIX.
-        vars['EXT_SUFFIX'] = _imp.extension_suffixes()[0]
-    except IndexError:
-        pass
+        # e.g., "_d.cp313-win_amd64.pyd"
+        vars['EXT_SUFFIX'] = extension_suffixes[0]
+
+        # e.g., "cp313-win_amd64"
+        _, soabi, _ = vars['EXT_SUFFIX'].split('.')
+        vars['SOABI'] = soabi
+
+        # e.g., check for a "t" (for "threading") in SOABI
+        if re.match(r'cp\d+t', soabi):
+            vars['Py_NOGIL'] = 1
+
+    vars['LIBPL'] = _safe_realpath(os.path.join(get_config_var('installed_base'), 'libs'))
+    if hasattr(sys, 'dllhandle'):
+        import _winapi
+        dllhandle = _winapi.GetModuleFileName(sys.dllhandle)
+        vars['LIBRARY'] = os.path.basename(_safe_realpath(dllhandle))
```

Comment:
This is different from the proposed definition in https://github.com/python/cpython/issues/88402#issuecomment-1093916630 to better match POSIX platforms. `LIBRARY` is a filename like `libpython3.13.a` (not a directory).

Reply:
Yeah, that was probably a mistake. The semantics should be aligned.

cc @zooba

Reply:
The semantics I originally posted used `stem`, which excludes the `.dll` extension. I'm sure I had a need for it (it'd be convenient for calculating the import library name, for example), but it's easy enough to do that when we don't have any other dots in the filename.

Reply:
To be clear, I don't have a particular issue with exposing that information, it's just that the `LIBRARY` has a completely different semantic on POSIX, and it'd be confusing to have Windows differ.

I had a look and it doesn't seems there is any config variable exposing that on POSIX, so if you think it makes sense, we can expose that information in with a name that isn't being used on POSIX.

For reference, I uploaded the output of `python -m sysconfig` on my system (Arch Linux Python 3.11).
https://gist.github.com/FFY00/c7fd731e64729447bef1cd41fc731c60

Reply:
Yeah, looks like `LIBRARY` is for a static library and `LDLIBRARY` is for dynamic? Seems weird to have them both set, but I guess that's how the makefile is set up right now.

Reply:
Yep, exactly! On POSIX you can have builds with both libraries. The interpreter will usually still use the dynamic library in that case, but the static lib is useful for certain use-cases (eg. embedding).